tests/e2e/util.go

```diff
@@ -2538,3 +2538,22 @@ func getK8sNodeIP(node *v1.Node) string {
 	gomega.Expect(address).NotTo(gomega.BeNil(), "Unable to find IP for node: "+node.Name)
 	return address
 }
+
+//expectedAnnotation polls for the given annotation in pvc and returns error if its not present
+func expectedAnnotation(ctx context.Context, client clientset.Interface, pvclaim *v1.PersistentVolumeClaim, annotation string) error {
+	framework.Logf("Waiting for health annotation for pvclaim %v", pvclaim.Name)
+	waitErr := wait.Poll(healthStatusPollInterval, pollTimeout, func() (bool, error) {
+		framework.Logf("wait for next poll %v", healthStatusPollInterval)
+		pvc, err := client.CoreV1().PersistentVolumeClaims(pvclaim.Namespace).Get(ctx, pvclaim.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+		for pvcAnnotation := range pvc.Annotations {
```

Comment:
I looked at the above reference, you have taken an integer array, but in our case, it's a map.
generate a map and try the same it will work fine.
Ref: https://play.golang.org/p/LeJi-4G5YOz
Attaching my test run with the logs for your reference: https://gist.github.com/marunachalam/c8d0a83641cab23ea1c6ac4ba5c29b25

Reply:
My bad, thanks for clarifying